install.ps1

```diff
@@ -0,0 +1,601 @@
+.EXAMPLE
+  Usage:
+    Invoke-WebRequest ((New-Object System.Net.WebClient).DownloadString('https://get.rke2.io/install.ps1'))        
+    ./install.ps1 -Channel Latest -Mehtod Tar
+#>
+
+    [CmdletBinding()]
+param (
+    [Parameter()]
+    [ValidateSet("latest", "stable", "testing", "v1.20", "v1.19", "v1.18")]
+    [String]
+    $Channel = "stable",
+    [Parameter()]
+    [ValidateSet("choco", "tar")]
+    [String]
+    $Method = "tar",
+    [Parameter()]
+    [ValidateSet("agent")]
+    [String]
+    $Type = "agent",
+    [Parameter()]
+    [String]
+    $Version,
+    [Parameter()]
+    [String]
+    $TarPrefix = "C:\usr\local",
+    [Parameter()]
+    [Switch]
+    $Commit,
+    [Parameter()]
+    [String]
+    $AgentImagesDir = "C:\var\lib\rancher\rke2\agent\images",
+    [Parameter()]
+    [String]
+    $ArtifactPath = "",
+    [Parameter()]
+    [String]
+    $ChannelUrl = "https://update.rke2.io/v1-release/channels"
+)
+
+$ErrorActionPreference = 'Stop'
+
+function Write-InfoLog() {
+    Write-Info "[INFO] " "$@"
+}
+
+function Write-WarnLog() {
+    Write-Warn "[WARN] " "$@"
+}
+
+# fatal logs the given argument at fatal log level.
+function Write-FatalLog() {
+    Write-Output "[ERROR] " "$@"
+    if ([string]::IsNullOrEmpty($SUFFIX)) {
+        Write-Fatal "[ALT] Please visit 'https://github.com/rancher/rke2/releases' directly and download the latest rke2.$SUFFIX.tar.gz"
+    }
+    exit 1
+}
+
+# check_target_mountpoint return success if the target directory is on a dedicated mount point
+function Test-TargetMountpoint() {
+    $volumes = Get-WmiObject Win32_Volume
+    $found = $false
+    foreach ($volume in $volumes) {
+        if($volume.Contains("C:")) {
+            $found = $true
+        }
+    }    
+    return $found
+}
+
+function Test-TargetRo()
+{
+    New-Item -ItemType file $TarPrefix/.rke2-ro-test -and Remove-Item -Force $TarPrefix/.rke2-ro-test
+    if ($lastExitCode -ne "0")
+    {
+        return $true
+    }
+    return $false
+}
+
+# setup_env defines needed environment variables.
+function Set-Environment()
+{
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [String]
+        $DefaultTarPrefix
+    )
+    # --- bail if we are not administrator ---
+    $adminRole = [Security.Principal.WindowsBuiltInRole]::Administrator
+    $currentRole = [Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()
+    If (-NOT $currentRole.IsInRole($adminRole))
+    {
+        Write-FatalLog "You need to be administrator to perform this install"
+    }
+
+    Write-Host "Using $($Channel) channel of rke2 for installation"
+
+    # --- install tarball to /usr/local by default, except if /usr/local is on a separate partition or is read-only
+    # --- in which case we go into /opt/rke2.
+    if (Test-TargetMountpoint -and Test-TargetRo) {
+        $TarPrefix="C:\opt\rke2"
+        Write-Warn $DefaultTarPrefix + "is read-only or a mount point; installing to " + $TarPrefix
```

Comment:
Is this even a thing on Windows?

Reply:
We just ported applying the same logic. I think that is a good point and it was removed.